tests/ui/macros/rfc-3086-metavar-expr/syntax-errors.rs

```diff
@@ -137,6 +137,58 @@ macro_rules! unknown_metavar {
     //~| ERROR expected expression
 }
 
+macro_rules! wrong_concat_declarations {
+    ($ex:expr) => {
+        ${concat()}
+        //~^ ERROR expected identifier
+
+        ${concat(aaaa)}
+        //~^ ERROR expected comma
+
+        ${concat(aaaa,)}
+        //~^ ERROR expected identifier
+
+        ${concat(aaaa, 1)}
```

Comment:
why are numbers forbidden? `aaaa1` is a perfectly valid identifier...

Reply:
Literals is something I am planning to tackle in a following PR